packages/runtime/src/runtimes/react/components/Document.tsx

```diff
@@ -15,15 +15,14 @@ export const Document = memo(
     { document, fallback }: DocumentProps,
     ref: Ref<ElementImperativeHandle>,
   ): JSX.Element {
-    const isInitialData = match(document)
-      .with({ meta: { isInitialData: true } }, () => true)
-      .otherwise(() => false)
+    const isInitialData = document.rootElement == null
+    const rootElement = getDocumentRootElement(document)
```

Comment:
Isn't `getDocumentRootElement` a selector? I don't think we should "pierce" into modules under the `state` directory as this is meant to be encapsulated by the interface the wraps the Redux store. Also, selectors should be reserved to be used with Redux state, not as general getters when we have data that matches the state. We only want to have to change selectors when the state changes, and not couple it to this part of the codebase, which isn't dealing with that state module

Reply:
I noticed later that this isn't a selector. The naming is unfortunate, though, since it follows the same naming convention there. We also seem to be using it similar to selectors elsewhere (`Documents.getRootElement`). I think we should place this in a shared place that is not the `state` directory so that we can keep things in those modules encapsulated.

Reply:
Fair point, will address.

Reply:
Ended up making a separate directory within `state` to store this utility function. It felt weird to me to remove this from `state` when the function operates on a type that was defined in a `state` module - moreover, all usages (except for one) of this function are within `state` as well.